synapse/handlers/register.py

```diff
@@ -221,11 +221,18 @@ def register(
         # auto-join the user to any rooms we're supposed to dump them into
         fake_requester = create_requester(user_id)
 
-        # try to create the room if we're the first user on the server
+        # try to create the room if we're the first real user on the server. Note
+        # that an auto generated support user is not a real user and never be
+        # the user to create the room
         should_auto_create_rooms = False
-        if self.hs.config.autocreate_auto_join_rooms:
+        if (self.hs.config.autocreate_auto_join_rooms and
+                self.hs.config.support_user_id != user_id):
             count = yield self.store.count_all_users()
-            should_auto_create_rooms = count == 1
+            if self.hs.config.support_user_id is None:
```

Comment:
This is horrible, and assume that support user is created when the server is created - needs a rethink